### ISSUE TITLE
load() & create() issues iOS

### ISSUE THREAD

**Issue by bassemBMI** [Wizcorp/phonegap-plugin-wizCanvas#48]
Hi,
I try to use this plugin in iOS to solve "a video stream placed on top of the HTML view" issue.
but I have errors with create() or load() functions:

console.log(wizCanvas);
==> {name: "mainView", views: Object, throwError: function, create: function, remove: function, ...}

wizCanvas.create('mainView',{}, function(success){console.log('success',success);}, function(fail){console.log('fail', fail);});
==> fail - null

wizCanvas.show('mainView',{}, function(success){console.log('success',success);}, function(fail){console.log('fail', fail);});
==> TypeError: undefined is not an object (evaluating 'this.views[name].show')

Same error for wizCanvas.load();
==> TypeError: undefined is not an object (evaluating 'this.views[name].load')

Do I missed something in the configuration ?

**Comment by ronkorving** [comment]
If creation fails, there's no point in trying to show/load it (so the 2nd error is no surprise). I'm not sure why it would fail with a `null` error object though :-/

**Comment by jrouault** [comment]
`mainView` is a name reserved for the Cordova window. So you can not create a view with this name. Please try using another name.

**Comment by ronkorving** [comment]
It should in any case return an error response in the failure callback. It could have contained what you just said and would have avoided this issue.

**Comment by jrouault** [comment]
I agree with your statement. Contributions are more than appreciated too.

**Comment by ronkorving** [comment]
I don't know much about PhoneGap plugin development, but it seems the source of the solution would be here? https://github.com/Wizcorp/phonegap-plugin-wizCanvas/blob/master/platforms/ios/HelloCordova/Plugins/WizCanvasPlugin/WizCanvasPlugin.m#L98 (iOS)

@bassemBMI If you wanna make a PR to address this, that would be nice. Alternatively, we can first make an issue that describes this problem. There currently aren't many contributors on wizCanvas unfortunately.

**Comment by bassemBMI** [comment]
Thanks for your responses.
I tried with "mainView" because using any other names does not return any response (no success nor fail).

**Comment by jrouault** [comment]
The following code returns the success callback in the example app provided in the repo:
`wizCanvas.create('test',{}, function(success){console.log('success',success);}, function(fail){console.log('fail', fail);});`

Could you please provide more informations such as logs in XCode and your iOS version? Or a minimal project which have this issue?

**Comment by bassemBMI** [comment]
the iOS version is 8.1.1.
I will check the Xcode log later (I remove the plugin from the project right now).
console.log(wizCanvas()) return an object (described in my first comment), this mean that the plugin was loaded successfully ? If yes why the load function doesn't work (it supposed to look for index.js file and load it automatically if exist, but is not the case).

**Comment by bassemBMI** [comment]
Now the app crash after few sec at this line:
int retVal = UIApplicationMain(argc, argv, nil, @"AppDelegate");
Xcode log:
ibc++abi.dylib: terminating with uncaught exception of type NSException 

I just add this code:
wizCanvas.create('test',{}, function(success){console.log('success',success);}, function(fail){console.log('fail', fail);});

**Comment by jrouault** [comment]
I'm not sure what went wrong only from that. Did you add this code after cordova initialization?
Can you provide a minimal project where you have this issue?
